src/liballoc/collections/binary_heap.rs

```diff
@@ -884,7 +884,7 @@ impl<'a, T> Hole<'a, T> {
     }
 
     #[inline]
-    fn pos(&self) -> usize {
+    const fn pos(&self) -> usize {
```

Comment:
There's no const constructor for `Hole`

Reply:
These are all *internal* APIs.

Reply:
Then why constify them at all?

Reply:
To increase the code quality and to make the eventual constification of the public APIs that use these internal APIs easier once we give `const fn` more powers.

Reply:
How does an annotation that currently has no benefit "increase code quality"? And with respect to future changes: if and when we make some public API that relies on this `const`, doing this one-line change then is no significant additional burden. Furthermore, until that day it's just line noise that can hinder or get lost in refactorings.

Reply:
Summarizing some discord discussion: Using `const fn` to indicate lack of side effects, divorced from CTFE, is your preference but not consensus. Even besides the question of "are `const fn`s pure?", there's no coding style decision to indicate as many pure functions as possible with `const` if they are not relevant for CTFE (one could consider that annotation burden not worthwhile). Without such a style decision, it seems at best neutral to use it that way on one's own accord.

Reply:
Summarizing what I said on Discord; if @oli-obk wants me to change it, I will. But it seems to me a supreme waste of time.

Reply:
*Not* making a change is a supreme waste of time?